### PR TITLE
AdministratorName parameter should take value which is defined in enum 

### PR DESCRIPTION
**Pull request by SanjaMalesevic**

### CODE REVIEW

**Review by jaredmoo — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json [hunk 1]
```diff
@@ -77,7 +77,10 @@
             "in": "path",
             "description": "The administrator name.",
             "required": true,
-            "type": "string"
+            "type": "string",
+			"enum": [
+				"ActiveDirectory"
```
Comment (tjprescott) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: Does the change from string to enum result in a breaking change for some SDKs?

Comment (SanjaMalesevic) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: Currently, there is a difference in number of parameters for create/modify/delete methods in .NET and Python clients between MI and SQL DB. So, this change is made to aligned MI clients with SQL DB (they need to be auto-generated again based on updated json).
Jared has noticed in PR https://github.com/Azure/azure-cli/pull/10446 that there is parameter AdministratorName in method create_or_update which always takes the same value and that this parameter doesn't exist in clients for sql db.
I am not sure if it is a breaking change.

Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: Yes, it is a breaking change. However this is a new API and therefore right now is the time that break will have minimal impact. Also this change brings managed instance admin API into sync with server admin API.

**Comment by SanjaMalesevic** [comment]
Hi all, this is a gentle reminder to take look at this PR.
Thanks,
Sanja

**Comment by tjprescott** [comment]
@Azure/arm-api-review-board  please take a look.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json [hunk 2]
```diff
@@ -77,7 +77,10 @@
             "in": "path",
             "description": "The administrator name.",
             "required": true,
-            "type": "string"
+            "type": "string",
+			"enum": [
```
Comment (majastrz) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: >	 [](start = 0, length = 1)

Nit - tab-based indent vs. spaces everywhere else in the file.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json [hunk 3]
```diff
@@ -121,7 +124,10 @@
             "in": "path",
             "description": "The requested administrator name.",
             "required": true,
-            "type": "string"
+            "type": "string",
+			"enum": [
+				"ActiveDirectory"
+			]
```
Comment (majastrz) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: Enums should specify the `x-ms-enum` extension documented at [https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md#x-ms-enum](https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md#x-ms-enum).

Comment (majastrz) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json]: This applies to all enums that were added.

---
In reply to: [342237499](https://github.com/Azure/azure-rest-api-specs/pull/7377#discussion_r342237499) [](ancestors = 342237499)

**Review by majastrz — CHANGES_REQUESTED** [review]
I added a comment. Please take a look.

**Comment by majastrz** [comment]
Some of the CI/CD checks are failing on this PR also. Please take a look.

**Comment by SanjaMalesevic** [comment]
> Some of the CI/CD checks are failing on this PR also. Please take a look.

I saw from the error log that ModelValidation failed when trying to validate "examples" and "x-ms-examples", but there is no specific error here. @majastrz could you please help me to figure out what is wrong here?

**Comment by majastrz** [comment]
If you keep clicking links on the Details next to the failing check, you will eventually get to the Model Validation errors.

I copied the following from https://dev.azure.com/azure-sdk/public/_build/results?buildId=181455&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742&t=43324f1f-6724-5d61-785f-f0ac9fd256a1:
```
 error : 
operationId: ManagedInstanceAdministrators_CreateOrUpdate
scenario: Update administrator of managed instance
source: response
responseCode: '200'
severity: 0
code: OBJECT_ADDITIONAL_PROPERTIES
details:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - location
  message: 'Additional properties not allowed: location'
  path: ''
  title: '#/definitions/ManagedInstanceAdministrator'
  description: An Azure SQL managed instance administrator.
  position:
    line: 325
    column: 37
  url: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json
  directives: {}
  jsonPosition:
    line: 19
    column: 15
  jsonUrl: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceAdministratorUpdate.json
  jsonPath: ''

 error : 
operationId: ManagedInstanceAdministrators_CreateOrUpdate
scenario: Update administrator of managed instance
source: response
responseCode: '201'
severity: 0
code: OBJECT_ADDITIONAL_PROPERTIES
details:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - location
  message: 'Additional properties not allowed: location'
  path: ''
  title: '#/definitions/ManagedInstanceAdministrator'
  description: An Azure SQL managed instance administrator.
  position:
    line: 325
    column: 37
  url: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstanceAdministrators.json
  directives: {}
  jsonPosition:
    line: 33
    column: 15
  jsonUrl: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceAdministratorUpdate.json
  jsonPath: ''

##[error]Bash exited with code '1'.
Finishing: Model Validation
```

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
Please check on the build failure: https://dev.azure.com/azure-sdk/public/_build/results?buildId=181455&view=logs&j=d2a54e27-891b-5b91-4b1b-0c6d3d820755&t=4db5f656-3b59-5661-54ad-d4645c211b0f

**Comment by SanjaMalesevic** [comment]
I've fixed the examples and all checks have passed. Thanks for help.

**Pull request merged**